unliftio/src/UnliftIO/Exception.hs

```diff
@@ -155,6 +161,17 @@ catchAnyDeep = catchDeep
 catchJust :: (MonadUnliftIO m, Exception e) => (e -> Maybe b) -> m a -> (b -> m a) -> m a
 catchJust f a b = a `catch` \e -> maybe (liftIO (throwIO e)) b $ f e
 
+-- | A variant of 'catch' that catches both synchronous and asynchronous exceptions.
+--
+-- WARNING: This function (and other @*SyncOrAsync@ functions) is for advanced users. Most of the
+-- time, you probably want to use the non-@SyncOrAsync@ versions.
+--
+-- Before attempting to use this function, be familiar with the "Rules for async safe handling"
+-- section in
+-- [this blog post](https://www.fpcomplete.com/blog/2018/04/async-exception-handling-haskell/).
```

Comment:
Please add `@since` comments to each new export.

Reply:
https://github.com/fpco/unliftio/pull/80/commits/572f7ce07ae9008a6f77b66fb5ef2408d08ea7d0